Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -198,9 +200,20 @@ void DVDInterface::DTKStreamingCallback(DIInterruptType interrupt_type,
 
   if (interrupt_type == DIInterruptType::TCINT)
   {
+    // maximum_samples can only be 112 (if running at 32khz) or 168 (if running at 48khz), so just
+    // prepare stack space for the biggest possible case. This is true even for the GameCube's not
+    // quite correct sample rates.
+    constexpr size_t MAX_POSSIBLE_SAMPLES = 168;
+
     // Send audio to the mixer.
-    std::vector<s16> temp_pcm(m_pending_samples * 2, 0);
-    ProcessDTKSamples(&temp_pcm, audio_data);
+    std::array<s16, MAX_POSSIBLE_SAMPLES * 2> temp_pcm;
+    ASSERT(m_pending_samples <= MAX_POSSIBLE_SAMPLES);
+    const size_t samples_processed =
+        ProcessDTKSamples(temp_pcm.data(), m_pending_samples, audio_data);
```

Comment:
min(m_pending_samples, MAX_POSSIBLE_SAMPLES)

Reply:
Yeah, good point. I updated it.